vpc-site2site-vpn/tf/main.tf

```diff
@@ -0,0 +1,265 @@
+provider "ibm" {
+  region             = "${var.region}"
+  ibmcloud_api_key   = "${var.ibmcloud_api_key}"
+  generation         = 1
+  softlayer_username = "${var.softlayer_username}"
+  softlayer_api_key  = "${var.softlayer_api_key}"
+}
+
+locals {
+  BASENAME = "${var.prefix}-vpc"
+
+  user_data_cloud = <<EOF
+#!/bin/bash
+apt-get update
+apt-get install -y nodejs npm
+EOF
+}
+
+data "ibm_resource_group" "all_rg" {
+  name = "${var.resource_group_name}"
+}
+
+resource "ibm_is_vpc" "vpc" {
+  name           = "${var.vpc_name == "" ? local.BASENAME : var.vpc_name}"
+  resource_group = "${data.ibm_resource_group.all_rg.id}"
+}
+
+resource "ibm_is_public_gateway" "cloud" {
+  count = "${var.cloud_pgw?1:0}"
+  vpc   = "${ibm_is_vpc.vpc.id}"
+  name  = "${local.BASENAME}-${var.zone}-pubgw"
+  zone  = "${var.zone}"
+}
+
+resource "ibm_is_public_gateway" "bastion" {
+  count = "${var.bastion_pgw?1:0}"
+  vpc   = "${ibm_is_vpc.vpc.id}"
+  name  = "${local.BASENAME}-${var.zone}-pubgw"
+  zone  = "${var.zone}"
+}
+
+resource "ibm_is_subnet" "cloud" {
+  name                     = "${local.BASENAME}-cloud-subnet"
+  vpc                      = "${ibm_is_vpc.vpc.id}"
+  zone                     = "${var.zone}"
+  public_gateway           = "${join("", ibm_is_public_gateway.cloud.*.id)}"
+  total_ipv4_address_count = 256
+}
+
+# bastion subnet and instance values needed by the bastion module
+resource "ibm_is_subnet" "bastion" {
+  name                     = "${local.BASENAME}-bastion-subnet"
+  vpc                      = "${ibm_is_vpc.vpc.id}"
+  zone                     = "${var.zone}"
+  total_ipv4_address_count = 256
+}
+
+data "ibm_is_image" "os" {
+  name = "${var.cloud_image_name}"
+}
+
+data "ibm_is_ssh_key" "sshkey" {
+  name = "${var.ssh_key_name}"
+}
+
+locals {
+  bastion_ingress_cidr    = "0.0.0.0/0" # DANGER: cidr range that can ssh to the bastion when maintenance is enabled
+  maintenance_egress_cidr = "0.0.0.0/0" # cidr range required to contact software repositories when maintenance is enabled
+}
+
+module bastion {
+  source            = "../../vpc-secure-management-bastion-server/tfmodule"
+  basename          = "${local.BASENAME}"
+  ibm_is_vpc_id     = "${ibm_is_vpc.vpc.id}"
+  zone              = "${var.zone}"
+  remote            = "${local.bastion_ingress_cidr}"
+  profile           = "${var.profile}"
+  ibm_is_image_id   = "${data.ibm_is_image.os.id}"
+  ibm_is_ssh_key_id = "${data.ibm_is_ssh_key.sshkey.id}"
+  ibm_is_subnet_id  = "${ibm_is_subnet.bastion.id}"
```

Comment:
@dprosper we should be passing the "resource_group_id" where to create the bastion resources.

Reply:
@l2fprod I think we should have a separate task for this, the bastion module does not currently accept a resource group variable.  Changing it for this script will certainly impact other scripts that leverage that module.

Reply:
@dprosper yes - I did create those tasks today https://github.com/IBM-Cloud/vpc-tutorials/issues/63